### PR TITLE
[docker] Teach Hail Ubuntu Image how to resiliently install apt packages

### PR DESCRIPTION
**Pull request by danking**
This PR is stacked on #9593. The key files to look at are in `docker/hail-ubuntu`. I introduced `hail-apt-get-install` which packages up the `apt-get update` and the removal of temporary files. I also set the number of package-download retries to 5.

### CODE REVIEW

**Comment by jigold** [comment]
I took a quick look and it looks fine, but I want to wait until the first goes in as it's a lot of lines.

**Comment by jigold** [comment]
This needs a rebase. Ping me when it's ready.

**Comment by danking** [comment]
@jigold ready; I also fixed the recently added `echo` image.

File: site/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
+FROM {{ hail_ubuntu_image.image }}
 
-RUN apt-get update && \
-    apt-get install -y nginx curl python-minimal && \
-    rm -rf /var/lib/apt/lists/*
+RUN hail-apt-get-install \
+  nginx \
+  curl \
+  python-minimal \
+  libnginx-mod-http-subs-filter
```
Comment (jigold) [inline site/Dockerfile:7]: Where did this one come from?

File: letsencrypt/Dockerfile [hunk 2]
```diff
@@ -1,15 +1,10 @@
 FROM {{ hail_ubuntu_image.image }}
 
-RUN apt-get update -y
-
 # get add-apt-repository
-RUN apt-get update -y && \
-    apt-get install -y nginx software-properties-common && \
+RUN hail-apt-get-install nginx software-properties-common && \
     bash -c 'DEBIAN_FRONTEND=noninteractive apt-get install -y tzdata' && \
```
Comment (jigold) [inline letsencrypt/Dockerfile:5]: Should we have not converted this image to be based on Ubuntu? Or was it always on ubuntu/base_image. Looking at the DEBIAN_FRONTEND variable

**Review by jigold — CHANGES_REQUESTED** [review]

File: site/Dockerfile (hunk 1 above)
Comment (danking) [reply site/Dockerfile:7]: Ubuntu nginx does not include this nginx module which we use. Debian nginx included it by default.

File: letsencrypt/Dockerfile (hunk 2 above)
Comment (danking) [reply letsencrypt/Dockerfile:5]: Ubuntu is built on top of Debian, so it leaks into ubuntu things. Here, we're telling apt-get, a Debian package manager that Ubuntu adopted, to be non-interactive.

This was originally using the ubuntu image.

**Review by jigold — APPROVED** [review]

**Comment by jigold** [comment]
Approved. Is there any reason to not merge. The Stacked PR label is still on.

**Comment by danking** [comment]
whoops, nope

**Comment by tpoterba** [comment]
commit went in as "wip" :(

**Comment by tpoterba** [comment]
we can add a check in CI to look at commit messages and reject if the only difference between the branch and main is a single commit that's fewer than X chars or somethign

**Comment by danking** [comment]
That's a great suggestion

**Comment by danking** [comment]
Actually, I think the right fix is for CI to grab the PR message and set it as the merge message

**Comment by tpoterba** [comment]
is that something the GitHub API supports? If so, that definitely sounds like the right thing!

**Comment by danking** [comment]
yeah, "merge_message" https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#merge-a-branch

**Pull request merged**